Add landing page layout

bounchich1 opened the pull request:
Implement secondary style of button component.
Add landing.png file, because svg is too heavy and complex.

bounchich1 committed:
Add landing page layout

Implement secondary style of button component.
Add landing.png file, because svg is too heavy and complex.

vbishel reviewed (changes requested):
### 1. BEM methodology 

In BEM methodology there is **no such** namings as container__container2__container3. There **must** be at most one double underscore in each class name. Double underscore is a separator between your _main_ container name (in your Landing.tsx it is landing) and element's own class name
If element's name consists of multiple words, separate them with single hyphen (-) 

For example:
- landing__left__info -> landing__left-info (main-container-name__own-class-name)
- landing__left__info__heading__text -> landing__heading-text
- landing__left__info__buttons -> landing__buttons

### 2. Minor mistakes

- In your _Landing.tsx_, the "landing" class, which is supposed to be the main container class of the whole page, nested inside "container" class. It is not correct - better to swap them and nest "container" inside "landing", and name it something like landing__container
- There is no need of applying bold styling to primary button text. It looks quite weird; just delete text-bold and leave it as is
- Navbar sign in button looks too large, make it smaller
- The page is too wide, make it thinner; look at spacing between text and image in figma design file, and take it as a reference. Don't use spacings from page edges as references, keep in mind that pages in figma are 1440px x 1024px, while computer screens are in most cases 1920px x 1080px. Check attached image for references

![Скриншот 11-10-2024 172031](https://github.com/user-attachments/assets/e5036a51-3a7f-4d6c-8046-1d451f0cc03d)

bounchich1 committed:
Change layout and code style

Change scss class structure according to BEM.
Remove bold font-style from primary button.
Change size of navbar "Sign In" button.
Make landing page thinner.

vbishel reviewed (changes requested):
In BEM methodology we typically don't include secondary containers name in own class names, follow the general rule:

className="main-container__own-class-name"

If you have two types of buttons that share some common properties you can use modificators

className="main-container__own-class-name main-container__own-class-name--modificator-name"

Ex:
- landing__container-left-info -> landing__left-info
- landing__container-left-info-heading-text -> landing__heading-text
- landing__container-left-info-heading -> landing__heading
- landing__container-left-info-buttons -> landing__buttons
- landing__container-left-info-buttons-secondary -> landing__button landing__button--secondary

bounchich1 committed:
Change scss class names according to BEM

vbishel reviewed (changes requested):
1. landing.scss file and the overall class structure is hardly readable. Don't nest so many classes inside one another, just create one single class for each element and put attributes in it. Why nest &-width inside &-buttons insde &__heading when you can just write simple landing__buttons class containing everything you need you need in it?
Please rewrite the scss file so you don't have any nesting that looks like &-classname

2. Also, if you are using modificator (landing__buttons--secondary) you should write className likes this: "landing__buttons landing__buttons--secondary" for your landing__buttons class to apply, instead there would be only "landing__buttons--secondary" applied

3. Furthermore, you are again referencing your secondary containers in your class names, for example: landing__left-logo should be just landing__logo, landing__heading-buttons-text should be just landing__buttons-text

bounchich1 committed:
Restructure of landing.scss

Split nested classes.
Make more clear the logic of buttons class.
Rename one of the buttons class to interface in case it is not nested inside left or right class.
Merge left and right classes into __content class, which has selectors --left and --right.
Make image sizing classes proper naming.
Delete duplicates.

bounchich1 committed:
Rename interface to navbar

Fix minor bug.
Add Link component for contact us button.
